### PR TITLE
Enable using -s EXIT_RUNTIME=0/1 when paired with -s PROXY_TO_PTHREAD=1 option.

### PR DESCRIPTION
**Pull request by juj**
Enable using -s EXIT_RUNTIME=1 when paired with -s PROXY_TO_PTHREAD=1. Previously PROXY_TO_PTHREAD=1 always implied EXIT_RUNTIME=0, and in this build mode, EXIT_RUNTIME=0 would specify the behavior of what the main browser thread would do after calling __proxy_main(). I.e. the main browser thread should not exit when the internal __proxy_main() function has been called. Users did not have a possibility to do -s PROXY_TO_PTHREAD=1 -s EXIT_RUNTIME=1 builds at all.

After this change, -s EXIT_RUNTIME=0/1 now applies to the main() function instead, like one would expect: specifying -s EXIT_RUNTIME=1 means that the application will quit immediately when the proxied main() returns. Specifying -s EXIT_RUNTIME=0 means that the application will continue on executing after the proxied main() returns.

### CODE REVIEW

**Comment by juj** [comment]
Hmm, the CI Failed in lsan tests, but oddly when I run locally, the same lsan tests fail in current incoming as well even without this PR. Rebased on top of latest incoming to check if the failure repeats...

**Comment by juj** [comment]
Oh, hmm, while debugging this, I realized a better way to do this altogether.

Changed this PR to actually add support for `-s EXIT_RUNTIME=0/1` in pthreads. @kripken: want to take another look?

File: system/lib/pthread/library_pthread.c [hunk 1]
```diff
@@ -937,7 +937,6 @@ int EMSCRIPTEN_KEEPALIVE proxy_main(int argc, char** argv) {
       // Proceed by running main() on the main browser thread as a fallback.
       return __call_main(_main_arguments.argc, _main_arguments.argv);
     }
-    EM_ASM(noExitRuntime = true);
```
Comment (kripken) [inline system/lib/pthread/library_pthread.c]: how is this change related to PROXY_TO_PTHREAD?

File: src/library_pthread.js [hunk 2]
```diff
@@ -408,6 +408,7 @@ var LibraryPThread = {
               if (detached) {
                 PThread.returnWorkerToPool(worker);
               }
+#if EXIT_RUNTIME // If building with -s EXIT_RUNTIME=0, no thread will post this message, so don't even compile it in.
```
Comment (sbc100) [inline src/library_pthread.js]: What about if the application actually calls "exit()"?   Or does `-s EXIT_RUNTIME=0` keep the program alive in that case too?

File: system/lib/pthread/library_pthread.c (hunk 1 above)
Comment (juj) [reply system/lib/pthread/library_pthread.c]: This EM_ASM line is what used to keep the runtime alive in PROXY_TO_PTHREAD build mode.

File: src/library_pthread.js (hunk 2 above)
Comment (juj) [reply src/library_pthread.js]: If you look at the implementation of `exit()`, you can see that it is a no-op in `-s EXIT_RUNTIME=0` builds. (Also calling `exit()` never posted a `exitProcess` message)

File: system/lib/pthread/library_pthread.c (hunk 1 above)
Comment (kripken) [reply system/lib/pthread/library_pthread.c]: I see, thanks!

**Comment by kripken** [comment]
Btw, it might be useful to do less force-pushes in PRs, because it would allow seeing the changes over time (and we squash them at the end anyhow). I'm not sure how the code changed here in the new version?

**Comment by juj** [comment]
Ah sorry - yeah, out of habit I tend to force push since we used to have a workflow to preserve commit history from PRs.

The original version of the code just force-enabled EXIT_RUNTIME=0 when PROXY_TO_PTHREAD=1 was in effect. (i.e. just did from command line the same thing that the EM_ASM() block achieved), but this new version instead changes the meaning of EXIT_RUNTIME to not apply to the main browser thread, but to the proxied main() thread, so it has same functional meaning as when used in single-threaded builds.

**Comment by sbc100** [comment]
> Btw, it might be useful to do less force-pushes in PRs, because it would allow seeing the changes over time (and we squash them at the end anyhow). I'm not sure how the code changed here in the new version?

BTW, for my changes I follow a rebase workflow locally so I almost always have to force push.     I can preserve the original commit if you prefer but I also sometime like to squash locally too.. I'll try not to do that with this repo if you prefer.
